main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java

```diff
@@ -639,17 +639,23 @@ private void switchTileLayer(final MapSource newSource) {
         // Create new render layer, if mapfile exists
         final ITileLayer oldLayer = this.tileLayer;
         ITileLayer newLayer = null;
+        final MapZoomControls zoomControls = mapView.getMapZoomControls();
         if (newSource instanceof MapsforgeMapProvider.OfflineMapSource) {
             this.mapView.getModel().displayModel.setFixedTileSize(0);
             final File mapFile = NewMap.getMapFile();
             if (mapFile != null && mapFile.exists()) {
-                newLayer = new RendererLayer(tileCache, new MapFile(mapFile), this.mapView.getModel().mapViewPosition, false, true, false, AndroidGraphicFactory.INSTANCE);
-
+                final MapFile mapDataStore = new MapFile(mapFile);
+                newLayer = new RendererLayer(tileCache, mapDataStore, this.mapView.getModel().mapViewPosition, false, true, false, AndroidGraphicFactory.INSTANCE);
+                final MapFileInfo mapFileInfo = mapDataStore.getMapFileHeader().getMapFileInfo();
+                zoomControls.setZoomLevelMax(mapFileInfo.zoomLevelMax);
+                zoomControls.setZoomLevelMin(mapFileInfo.zoomLevelMin);
             }
         } else {
             this.mapView.getModel().displayModel.setFixedTileSize(256);
             if (newSource.getNumericalId() == MapsforgeMapProvider.MAPSFORGE_MAPNIK_ID.hashCode()) {
```

Comment:
Your change is perfectly okay. I'm just wondering if this condition can be not fulfilled at some point in time. And if we have to do something about it (cause I would expect one more "else" then). Basically, I would expect us to be able to convert the newSource to a provider instance without an if.

Reply:
Valid point. I'll have a look.